client/state/data-layer/wpcom/jetpack-install/index.js

```diff
@@ -0,0 +1,46 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	updateJetpackRemoteInstallError,
+	jetpackRemoteInstallComplete,
+} from 'state/jetpack-remote-install/actions';
+import { JETPACK_REMOTE_INSTALL } from 'state/action-types';
+
+export const installJetpackPlugin = ( { dispatch }, action ) => {
+	const { url, user, password } = action;
+	return dispatch(
+		http(
+			{
+				method: 'POST',
+				path: '/jetpack-install/' + url,
+				query: {
+					user,
+					password,
+				},
+			},
+			action
+		)
+	);
+};
+
+export const handleResponse = ( { dispatch }, { url }, data ) => {
+	if ( data.status ) {
+		return dispatch( jetpackRemoteInstallComplete( url ) );
+	}
+	dispatch( updateJetpackRemoteInstallError( url, data.error ) );
+};
+
+export const handleError = ( { dispatch }, { url } ) => {
+	dispatch( updateJetpackRemoteInstallError( url, 'API_ERROR' ) );
+};
+
+export default {
+	[ JETPACK_REMOTE_INSTALL ]: [
+		dispatchRequest( installJetpackPlugin, handleResponse, handleError ),
```

Comment:
Minor: Prefer new `dispatchRequestEx`, will be replacing `dispatchRequest`:

https://github.com/Automattic/wp-calypso/blob/0809df087e72918bb83cb500241406d35bb6a494/client/state/data-layer/wpcom-http/utils.js#L330-L343

For this case it's mostly shifting the args into an option object.